### ISSUE TITLE
Configurable tmp folder location

### ISSUE THREAD

**Issue by viamin** [stryker-mutator/stryker#336]
Currently Stryker creates a new dot folder in the current working directory which is not in a default temp location [https://github.com/stryker-mutator/stryker/blob/8e0e6b016b513a5fbd1f24fc7a08499ef3151dba/packages/stryker/src/utils/StrykerTempFolder.ts#L9](https://github.com/stryker-mutator/stryker/blob/8e0e6b016b513a5fbd1f24fc7a08499ef3151dba/packages/stryker/src/utils/StrykerTempFolder.ts#L9), which means every user of stryker needs to edit their .gitignore file.

It would be nice if this location could be configured in the stryker.conf.js file, and by default live in ./tmp/stryker instead of .stryker-tmp

**Comment by nicojs** [comment]
Thank you @viamin  for reporting this issue.

The position of your .stryker-tmp folder is indeed not ideal. We actually used your system's temp folder earlier, but we ran into a problem: `require` searches for node_modules relative to your current file. For example: `const chai = require('chai');` won't work if you copy your files to your systems temp folder. This is why we ended up with this solution.

In other words: your systems temp folder will only work for browser-based test runners like [karma](http://karma-runner.github.io/).

If we would add a configuration item for this it will be one more way users can make mistakes in configuring Stryker. It's already kind of hard to do. Would you still want this option?

**Comment by viamin** [comment]
I was not referring to system /tmp, but the project root ./tmp, which would be in the same directory as node_modules. I'm proposing a project with these files:
/path/to/project_root/tmp/stryker/ (replacing /path/to/project_root/.stryker-tmp/)
/path/to/project_root/node_modules
/path/to/project_root/package.json
etc.

**Comment by nicojs** [comment]
Ah i see what you mean. That should work indeed. Do you want to take a shot at implementing this? Would you only allow relative path locations? If we allow for the systems tmp folder for example, we should warn people about the node_modules.

**Comment by sharikovvladislav** [comment]
I think this one can be implemented after https://github.com/stryker-mutator/stryker/pull/361 or with it.

**Comment by avassem85** [comment]
Solution created in https://github.com/stryker-mutator/stryker/pull/475

**Comment by simondel** [comment]
We're closing this issue for now. If someone wants to fix this, let us know!
